File: Cisco/8MaximumDifferenceBetweenIncreasingElements.py
class Solution:
    def maximumDifference(self, nums: list[int]) -> int:
        N = len(nums)
        i = 0
        ans = -1

        for j in range(1, N):
            if nums[j] > nums[i]:
                ans = max(ans, nums[j] - nums[i])
            else:
                i = j

        return ans

File: Cisco/GitHubList/C++/8MaximumDifferenceBetweenIncreasingElements.cpp
#include <vector>

using namespace std;

class Solution
{
public:
    int maximumDifference(vector<int> &nums)
    {
        int iMin = 0, N = nums.size(), result = -1;
        for (int i = 1; i < N; ++i)
        {
            if (nums[i] > nums[iMin])
                result = max(result, nums[i] - nums[iMin]);
            else
                iMin = i;
        }
        return result;
    }
};